cloud/store/store.go

```diff
@@ -236,7 +236,15 @@ func (st *Store) Open(ctx context.Context) (err error) {
 			"snapshot_index", snapshotIndex(snapshotStore),
 			"last_applied_log_index", st.initialLastAppliedIndex)
 
-		if err := raft.RecoverCluster(st.raftConfig(), st, logCache, st.logStore, snapshotStore, st.transport, raft.Configuration{
+		// FSM passed to RecoverCluster has to be temporary one
+		// because it will be left in state shouldn't be used by the application.
+		if err := raft.RecoverCluster(st.raftConfig(), &Store{
```

Comment:
Question: In which state is the store left after recovery that makes it unusable ? I'm curious here because as someone who doesn't know the internals of it, I would expect it to be a working store after recovery. What is the issue we encounter ?

Reply:
this FSM is used for snapshots ops only, it reads all the current state for this node, creates a snapshot with the
supplied configuration, and then truncates the Raft log. 

this addresses the documentation of `RecoverCluster()` https://github.com/hashicorp/raft/blob/main/api.go#L291 

`RecoverCluster()` is actually a dangerous op and we were discussing that we gonna revisit it and make it manual to make sure we have consistent peers config because if one of the peers specially after operation like this which gonna force creating snapshot() and has stale/conflicted log and won the election could conflict the whole cluster, therefore we should have consistent config across the cluster